File: Hjemmesentral/app/src/main/java/com/homecentral/jrs/hjemmesentral/model/yr/forecast/Time.java
package com.homecentral.jrs.hjemmesentral.model.yr.forecast;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;


@Root(name = "time", strict = false)
public class Time {

    @Attribute(name = "from")
    private String from;

    @Attribute(name = "to")
    private String to;

    @Attribute(name = "period")
    private String period;

    @Element(name = "symbol", required = false)
    private Symbol symbol;

    @Element(name = "precipitation", required = false)
    private Precipitation precipitation;

    @Element(name = "windDirection", required = false)
    private WindDirection windDirection;

    @Element(name = "windSpeed", required = false)
    private WindSpeed windSpeed;

    @Element(name = "temperature", required = false)
    private Temperature temperature;

    @Element(name = "pressure", required = false)
    private Pressure pressure;

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }

    public String getPeriod() {
        return period;
    }

    public void setPeriod(String period) {
        this.period = period;
    }

    public Symbol getSymbol() {
        return symbol;
    }

    public void setSymbol(Symbol symbol) {
        this.symbol = symbol;
    }

    public Precipitation getPrecipitation() {
        return precipitation;
    }

    public void setPrecipitation(Precipitation precipitation) {
        this.precipitation = precipitation;
    }

    public WindDirection getWindDirection() {
        return windDirection;
    }

    public void setWindDirection(WindDirection windDirection) {
        this.windDirection = windDirection;
    }

    public WindSpeed getWindSpeed() {
        return windSpeed;
    }

    public void setWindSpeed(WindSpeed windSpeed) {
        this.windSpeed = windSpeed;
    }

    public Temperature getTemperature() {
        return temperature;
    }

    public void setTemperature(Temperature temperature) {
        this.temperature = temperature;
    }

    public Pressure getPressure() {
        return pressure;
    }

    public void setPressure(Pressure pressure) {
        this.pressure = pressure;
    }
}

File: Hjemmesentral/app/src/main/java/com/homecentral/jrs/hjemmesentral/model/yr/forecast/Time.kt
package com.homecentral.jrs.hjemmesentral.model.yr.forecast

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "time", strict = false)
class Time @JvmOverloads constructor(
    @field:Attribute(name = "from") var from: String = "",
    @field:Attribute(name = "to") var to: String = "",
    @field:Attribute(name = "period") var period: String = "",
    @field:Element(name = "symbol", required = false) var symbol: Symbol? = null,
    @field:Element(name = "precipitation", required = false) var precipitation: Precipitation? = null,
    @field:Element(name = "windDirection", required = false) var windDirection: WindDirection? = null,
    @field:Element(name = "windSpeed", required = false) var windSpeed: WindSpeed? = null,
    @field:Element(name = "temperature", required = false) var temperature: Temperature? = null,
    @field:Element(name = "pressure", required = false) var pressure: Pressure? = null
)
